substrate/client/db/src/lib.rs

```diff
@@ -1239,6 +1239,24 @@ impl<Block: BlockT> Backend<Block> {
 
 		let offchain_storage = offchain::LocalStorage::new(db.clone());
 
+		let shared_trie_cache = config.trie_cache_maximum_size.map(|maximum_size| {
+			let system_memory = sysinfo::System::new_all();
+			let used_memory = system_memory.used_memory();
+			let total_memory = system_memory.total_memory();
+
+			if maximum_size as u64 > total_memory - used_memory {
+				return Err(sp_blockchain::Error::Backend(
+					format!(
+						"Not enough memory to initialize shared trie cache. Cache size: {} bytes. System memory: used {} bytes, total {} bytes",
+						maximum_size, used_memory, total_memory,
+					)
+				));
+			}
+
+			info!("Initializing shared trie cache with size {} bytes, {}% of total memory", maximum_size, (maximum_size as f64 / total_memory as f64 * 100.0));
```

Comment:
```suggestion
			debug!("Initializing shared trie cache with size {} bytes, {}% of total memory", maximum_size, (maximum_size as f64 / total_memory as f64 * 100.0));
```

Not really seeing the benefit for printing this to info.